server/routerlicious/packages/services-core/src/runWithRetry.ts

```diff
@@ -148,10 +148,10 @@ export async function requestWithRetry<T>(
                     Lumberjack.info(`Succeeded in executing ${callName} with ${retryCount} retries`, telemetryProperties);
                 }
             } catch (error) {
-                Lumberjack.error(`Error running ${callName}: retryCount ${retryCount}`, telemetryProperties, error);
                 if (onErrorFn !== undefined) {
                     onErrorFn(error);
                 }
+                Lumberjack.error(`Error running ${callName}: retryCount ${retryCount}`, telemetryProperties, error);
```

Comment:
Any reason why we are moving this log statement order?

Reply:
That log statement logs the error too, so error sanitization needs to be performed before that.

Reply:
Ah! Good point.

Reply:
Then, do we also need to do that in `runWithRetry` as well?
https://github.com/microsoft/FluidFramework/blob/94f6b19e1dcdc930f4ab0ea6052be6671191338f/server/routerlicious/packages/services-core/src/runWithRetry.ts#L24

Reply:
This is runWithRetry, right ? Did you mean a different one ?

Reply:
Sorry, my previous statement was confusing. The file contains two similar (but slightly different) implementations. One is called `requestWithRetry` - I think this is the one you updated - and the other one is called `runWithRetry`. The Mongo facade code internally calls `requestWithRetry`. But if it's a simple change to make, updating `runWithRetry` could potentially prevent other leaks in the future for its consumers

Reply:
ah, I see what you are saying. Makes sense. Will update.